controllers/k8ssandra/reaper.go

```diff
@@ -127,7 +127,10 @@ func (r *K8ssandraClusterReconciler) reconcileReaper(
 		if err := remoteClient.Get(ctx, reaperKey, actualReaper); err != nil {
 			if errors.IsNotFound(err) {
 				logger.Info("Creating Reaper resource")
-				if err := remoteClient.Create(ctx, desiredReaper); err != nil {
+				if err := controllerutil.SetControllerReference(actualDc, desiredReaper, r.Scheme); err != nil {
```

Comment:
Question: why do we do this inside reconcileReaperSchema? Seems like an odd place to reconcile k8s resources.

Reply:
It is inside `reconcileReaper()`, which handles the creation of the `Reaper` k8s resource.